Python/tier2_redundancy_eliminator_bytecodes.c

```diff
@@ -197,6 +197,17 @@ dummy_func(void) {
         }
     }
 
+    op(_BINARY_OP_ADD_UNICODE, (left, right -- res)) {
+        if (is_const(left) && is_const(right)) {
+            PyObject *temp = PyUnicode_Concat(get_const(left), get_const(right));
+            ERROR_IF(temp == NULL, error);
+            OUT_OF_SPACE_IF_NULL(res = sym_new_const(ctx, temp));
```

Comment:
This worries me about memory leakage. If we have a valid concatenation result but the `sym_new_const()` ran out of space, we leak the result. Even worse, that is documented as taking a borrowed reference. I *think* that's what we normally call *stealing* a reference? *Borrowed* references are normally return values (indicating that the caller shouldn't DECREF it) but I believe for arguments we don't use borrowed.

One way out (assuming the intention is for `sym_new_const()` to take ownership of its `const_val` argument) is to make that function `DECREF(const_val)` when it returns NULL.